src/tree/mod.rs

```diff
@@ -406,7 +407,7 @@ impl Tree {
             true,
         );
         self.pages
-            .prepend(new_root_pid, new_root_cas_key, new_root)
+            .replace(new_root_pid, new_root_cas_key, vec![new_root])
```

Comment:
replace is used as a signifier that a new page is being written over a previous one completely, or it's the first time we're using a page